src/webgpu/util/compare.ts

```diff
@@ -245,6 +272,12 @@ export function deserializeComparator(data: SerializedComparator): Comparator {
     case 'anyOf': {
       return anyOf(...data.data.map(e => deserializeExpectation(e)));
     }
+    case 'skipUndefined': {
+      if (data.data.length === 0) {
+        return skipUndefined(undefined);
+      }
+      return skipUndefined(deserializeExpectation(data.data[0]));
+    }
   }
-  throw `unhandled comparator kind: ${data.kind}`;
```

Comment:
The usage of data.kind is now invalid, because the type is 'anyOf' | 'skipUndefined', and not string. When 'anyOf' was the only option I think TS was able to infer the string type.